src/GraphQL.Conventions/Execution/UserContextExtensions.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System.Collections.Generic;
+using GraphQL.Execution;
+using GraphQL.Types;
+
+namespace GraphQL.Conventions.Execution
+{
+    public static class UserContextExtensions
+    {
+        public static IUserContext GetUserContext(this ExecutionContext context) => GetValue<IUserContext>(context.UserContext);
+        public static IUserContext GetUserContext(this ResolveFieldContext context) => GetValue<IUserContext>(context.UserContext);
+        public static IUserContext GetUserContext<T>(this ResolveFieldContext<T> context) => GetValue<IUserContext>(context.UserContext);
```

Comment:
Maybe it worth changing ResolveFieldContext to IResolveFieldContext everywhere in this file?